fastlane_bot/events/async_event_update_utils.py

```diff
@@ -361,13 +361,13 @@ def get_pool_contracts(mgr: Any) -> List[Dict[str, Any]]:
     contracts = []
     for add, en, event, key, value in mgr.pools_to_add_from_contracts:
         exchange_name = mgr.exchange_name_from_event(event)
-        ex = exchange_factory.get_exchange(key=exchange_name, cfg=mgr.cfg, exchange_initialized=False)
+        ex = mgr.exchanges[exchange_name]
```

Comment:
The code which you've removed:
```
exchange_factory.get_exchange(key=exchange_name, cfg=mgr.cfg, exchange_initialized=False)
```
Is used in a few other places.

Why was it used like this here, and in each one of the other places?

Are you sure that changing it only here is the correct thing to do?

The more general question to ask here - why is the `exchange_factory` object not just a simple dictionary? Finding the exact reason for this will probably answer all the other questions above.

Reply:
Thank you! I updated this in the other places that I found it too. 

Now exchange_factory.get_exchange is only used once on startup, and other times it is referenced from the dictionary stored in the base manager.

Reply:
I've found 4 places in the code where we call function `get_exchange`, and pass to it more than just the exchange name:
- 2 in file async_event_update_utils.py
- 1 in file async_utils.py
- 1 in file base.py

I'm assuming that the one in file base.py is the one that your refer to as "once on startup".

Have you addressed all the others?

Reply:
Yes - I changed this in each location except for base.py.